api/handlers.go

```diff
@@ -55,7 +59,79 @@ func (s *Server) Pulse(ctx echo.Context, pulseNumber server.PulseNumberPathParam
 }
 
 func (s *Server) JetDropsByPulseNumber(ctx echo.Context, pulseNumber server.PulseNumberPathParam, params server.JetDropsByPulseNumberParams) error {
-	panic("implement me")
+	if !pulse.IsValidAsPulseNumber(int(pulseNumber)) {
+		response := server.CodeValidationError{
+			Code:               NullableString("400"),
+			Description:        NullableString("Invalid pulse number"),
+			Link:               nil,
+			Message:            nil,
+			ValidationFailures: nil,
+		}
+		return ctx.JSON(http.StatusBadRequest, response)
+	}
+
+	limit, offset, err := checkLimitOffset(params.Limit, params.Offset)
+	if err != nil {
+		response := server.CodeValidationError{
+			Code:               NullableString("400"),
+			Description:        NullableString("Invalid limit or offset query parameters"),
+			Link:               nil,
+			Message:            NullableString(err.Error()),
+			ValidationFailures: nil,
+		}
+		return ctx.JSON(http.StatusBadRequest, response)
+	}
+
+	jetDropID, err := checkJetDropID(params.FromJetDropId)
+	if err != nil {
+		response := server.CodeValidationError{
+			Code:               NullableString("400"),
+			Description:        NullableString("Invalid jet drop id in query parameters"),
+			Link:               nil,
+			Message:            NullableString(err.Error()),
+			ValidationFailures: nil,
+		}
+		return ctx.JSON(http.StatusBadRequest, response)
+	}
+	jetDrops, err := s.storage.GetJetDrops(
+		models.Pulse{PulseNumber: int(pulseNumber)},
+		jetDropID,
+		&limit,
+		&offset,
+	)
+	if err != nil {
+		s.logger.Error(err)
+		return ctx.JSON(http.StatusInternalServerError, struct{}{})
+	}
+
+	var result []server.JetDrop
+	for _, jetDrop := range jetDrops {
+		result = append(result, JetDropToAPI(jetDrop))
+
+	}
+	cnt := int64(len(result))
+	return ctx.JSON(http.StatusOK, server.JetDropsResponse{
+		Total:  &cnt,
```

Comment:
total is all of the raw in the DB. not in your DB query result. 
we can have a 1000 rows but you returned a 100, so total 1000, but the array contains 100 only

Reply:
fixed with new method